azureDatabaseUtils/src/main/java/bio/terra/workspace/azureDatabaseUtils/storage/AzureBlobStorage.java

```diff
@@ -0,0 +1,106 @@
+import java.io.BufferedOutputStream;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class AzureBlobStorage implements BlobStorage {
+  public AzureBlobStorage() {}
+
+  private static final Logger logger = LoggerFactory.getLogger(AzureBlobStorage.class);
+
+  /*
+  6/28/2023 - Terra's storage inside a billing project is organized as follows:
+  by default a billing project gets a single blob storage azure resource
+  each workspace that gets created inside of that billing project will get its own container inside the blob storage
+  the container will follow the naming of "sc-<workspaceId>"
+  all files that are uploaded into the workspace will be inside the appropriate workspace container
+  containers are restricted to have the same access as the users who have access to the workspace
+
+  backups will be stored inside the workspace container in a path determined by function GenerateBackupFilename
+  which exists within BackupService.java.
+   */
+  @Override
+  public void streamOutputToBlobStorage(
+      InputStream fromStream, String blobName, String workspaceId, String blobstorageDetails) {
+    BlobContainerClient blobContainerClient =
+        constructBlockBlobClient(workspaceId, blobstorageDetails);
+    // https://learn.microsoft.com/en-us/java/api/overview/azure/storage-blob-readme?view=azure-java-stable#upload-a-blob-via-an-outputstream
+    try (BufferedOutputStream blobOS =
+        new BufferedOutputStream(
+            blobContainerClient
+                .getBlobClient(blobName)
+                .getBlockBlobClient()
+                .getBlobOutputStream())) {
+      try (BufferedReader bufferedReader =
+          new BufferedReader(new InputStreamReader(fromStream, StandardCharsets.UTF_8))) {
+        int line;
+        while ((line = bufferedReader.read()) != -1) {
+          blobOS.write(line);
+        }
+      }
+    } catch (IOException ioEx) {
+      throw new LaunchProcessException("Error streaming output of child process", ioEx);
+    }
+  }
+
+  @Override
+  public void streamInputFromBlobStorage(
+      OutputStream toStream, String blobName, String workspaceId, String blobstorageDetails) {
+    BlobContainerClient blobContainerClient =
+        constructBlockBlobClient(workspaceId, blobstorageDetails);
+    try (toStream) {
+      blobContainerClient.getBlobClient(blobName).downloadStream(toStream);
+    } catch (IOException ioEx) {
+      throw new LaunchProcessException("Error streaming input to child process", ioEx);
+    }
+  }
+
+  private BlobContainerClient constructBlockBlobClient(
+      String workspaceId, String blobstorageDetails) {
+    // TODO: determine where blobstorageDetails should come from.
```

Comment:
seeking reviewer opinions on how to handle this. I've been manually testing this by passing it into the azureDatabaseUtils command. Ultimately, it seems like it should be passed in during CloneAllResourcesFlight (which will be worked on as part of [WM-2236](https://broadworkbench.atlassian.net/browse/WM-2236)).